viewstate/viewstate.py

```diff
@@ -38,18 +56,50 @@ def is_valid(self):
             return False
 
     def decode(self):
-        if not self.is_valid():
-            raise ViewStateException("Cannot decode invalid viewstate, bad preamble")
+        if self.decrypt:
+            if self.mac_mode == MAC.HMACMD5:
```

Comment:
Perhaps these `elif` blocks can be shrunk down to a single `dict` that holds all these values?